CONTRIBUTING.md

```diff
@@ -123,6 +123,46 @@ Represents a unique identifier for specific person. The good candidate is the Gi
  
 Please keep the name short and relevant.
 
+## Commit Message Structure Guideline
+
+Commits going to a master branch should stick to the Conventional Commits specification. This is a lightweight convention on the top of the commit messages. 
+Template:
+```
+<type>[optional scope]: <description>
+
+[optional body]
+
+[footer(s)]
+```
+Basic example:
+```
+fix(authentication): Introducing x509 as a form of authentication
```

Comment:
As an example shouldn't this be feat(authentication) ?

Reply:
yeah, good catch